crates/core_simd/src/vector.rs

```diff
@@ -673,18 +692,18 @@ where
     fn ne(&self, other: &Self) -> bool {
         // Safety: All SIMD vectors are SimdPartialEq, and the comparison produces a valid mask.
         let mask = unsafe {
-            let tfvec: Simd<<T as SimdElement>::Mask, LANES> = intrinsics::simd_ne(*self, *other);
+            let tfvec: Simd<<T as SimdElement>::Mask, N> = intrinsics::simd_ne(*self, *other);
             Mask::from_int_unchecked(tfvec)
         };
 
-        // Two vectors are non-equal if any lane tested true for vertical non-equality.
+        // Two vectors are non-equal if they are elementwise non-equal
```

Comment:
```suggestion
        // Two vectors are non-equal if any elements are non-equal when compared elementwise
```
`elementwise non-equal` implies to me producing a vector of results, not a scalar